packages/expo-modules-core/CHANGELOG.md

```diff
@@ -27,6 +27,7 @@
 
 ### 💡 Others
 
+- Deprecated `expo.modules.core.Promise`. ([#27369](https://github.com/expo/expo/pull/27369) by [@aleqsio](https://github.com/aleqsio))
```

Comment:
These changes should be made in a separate PR 😉 You're linking to a PR that has nothing to do with the deprecation.